### ISSUE TITLE
linter: invalid memory address or nil pointer dereference in prefer_return_this_type

### ISSUE THREAD

**Issue by GabbeV** [oxc-project/oxc#15633]
### What version of Oxlint are you using?

1.28.0

### What command did you run?

npx oxlint --type-aware repro.ts

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "categories": {
    "correctness": "off",
    "perf": "off",
    "style": "off",
    "nursery": "off",
    "pedantic": "off",
    "restriction": "off"
  },
  "plugins": [
    "typescript"
  ],
  "rules": {
    "typescript/prefer-return-this-type": "error"
  }
}
```

### What happened?

Operating system: Windows 10

repro.ts:
```ts
class A {
  static C = class B {
    d(): B {
      return new B();
    }
  };
}
```

error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x7ff601801052]

goroutine 27 [running]:
github.com/typescript-eslint/tsgolint/internal/rules/prefer_return_this_type.init.func1.2.1(0xc000098570?)
        /home/runner/work/tsgolint/tsgolint/internal/rules/prefer_return_this_type/prefer_return_this_type.go:81 +0x52
github.com/microsoft/typescript-go/internal/ast.ForEachReturnStatement.func1(0x7ff6018eb160?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/utilities.go:1158 +0x52
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:28
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:37
github.com/microsoft/typescript-go/internal/ast.(*Block).ForEachChild(0x7ff6018eb160?, 0xc000098570)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:3681 +0x67
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242
github.com/microsoft/typescript-go/internal/ast.ForEachReturnStatement.func1(0x58?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/utilities.go:1162 +0x9c
github.com/microsoft/typescript-go/internal/ast.ForEachReturnStatement(0xc0022fe540, 0xc00210f260)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/utilities.go:1166 +0x8e
github.com/typescript-eslint/tsgolint/internal/rules/prefer_return_this_type.init.func1.2(0xc0017ff970, 0xc0022e2980)
        /home/runner/work/tsgolint/tsgolint/internal/rules/prefer_return_this_type/prefer_return_this_type.go:69 +0x26e
github.com/typescript-eslint/tsgolint/internal/rules/prefer_return_this_type.init.func1.4(0xc0017ff970)
        /home/runner/work/tsgolint/tsgolint/internal/rules/prefer_return_this_type/prefer_return_this_type.go:124 +0x3c
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.6(0xaf, 0xc0017ff970)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:253 +0x5f
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0xc0017ff970)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:300 +0x42
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:28
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:37
github.com/microsoft/typescript-go/internal/ast.(*ClassLikeBase).ForEachChild(0xc0022e29b0, 0xc002eb00a0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:4164 +0x130
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0xc002ea00e8?, 0xc0022e2980?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x1c
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0xc0022e2980)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:315 +0x125
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:21
github.com/microsoft/typescript-go/internal/ast.(*PropertyDeclaration).ForEachChild(0xc0001f4a80, 0xc002eb00a0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:5983 +0xf1
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0xc002eb00ad?, 0xc0001f4a80?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x1c
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0xc0001f4a80)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:315 +0x125
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:28
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:37
github.com/microsoft/typescript-go/internal/ast.(*ClassLikeBase).ForEachChild(0xc002113c58, 0xc002eb00a0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:4164 +0x130
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x7ff6012c0108?, 0xc002113c20?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x1c
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0xc002113c20)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:315 +0x125
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:28
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:37
github.com/microsoft/typescript-go/internal/ast.(*SourceFile).ForEachChild(0xc002342308, 0xc002eb00a0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:10862 +0x9b
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x7ff6018aafe0?, 0xc002eac600?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x1c
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1()
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:323 +0x11e
github.com/microsoft/typescript-go/internal/core.(*parallelWorkGroup).Queue.func1()
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/core/workgroup.go:40 +0x13
sync.(*WaitGroup).Go.func1()
        /opt/hostedtoolcache/go/1.25.0/x64/src/sync/waitgroup.go:239 +0x4a
created by sync.(*WaitGroup).Go in goroutine 1
        /opt/hostedtoolcache/go/1.25.0/x64/src/sync/waitgroup.go:237 +0x73
Error running tsgolint: "exit status: exit code: 2"
```
